numbers/src/test/java/org/dicio/numbers/lang/fr/FrenchFormatter.java

```diff
@@ -0,0 +1,236 @@
+                result.append("virgule ");
+                result.append(NUMBER_NAMES.get(numberLong % 10));
+            } else if (numberLong % 10 == 0 || numberLong % 100 < 20) {
+                // 1960 => nineteen sixty; 1911 => nineteen eleven
+                result.append(NUMBER_NAMES.get(numberLong % 100));
+            } else {
+                // 1961 => nineteen sixty one
+                result.append(NUMBER_NAMES.get(numberLong % 100 - numberLong % 10));
+                result.append(" ");
+                result.append(NUMBER_NAMES.get(numberLong % 10));
+            }
+
+            return result.toString();
+        }
+
+        if (!ordinal && NUMBER_NAMES.containsKey(numberLong)) {
+            if (number > 90) {
+                result.append("one ");
+            }
+            result.append(NUMBER_NAMES.get(numberLong));
+
+        } else if (shortScale) {
+            boolean ordi = ordinal && numberIsWhole; // not ordinal if not whole
+            final List<Long> groups = Utils.splitByModulus(numberLong, 1000);
+            final List<String> groupNames = new ArrayList<>();
+            for (int i = 0; i < groups.size(); ++i) {
+                final long z = groups.get(i);
+                if (z == 0) {
+                    continue; // skip 000 groups
+                }
+                String groupName = subThousand(z, i == 0 && ordi);
+
+                if (i != 0) {
+                    final long magnitude = Utils.longPow(1000, i);
+                    if (ordi) {
+                        // ordi can be true only for the first group (i.e. at the end of the number)
+                        if (z == 1) {
+                            // remove "one" from first group (e.g. "one billion, millionth")
+                            groupName = ORDINAL_NAMES_SHORT_SCALE.get(magnitude);
+                        } else {
+                            groupName += " " + ORDINAL_NAMES_SHORT_SCALE.get(magnitude);
+                        }
+                    } else {
+                        groupName += " " + NUMBER_NAMES_SHORT_SCALE.get(magnitude);
+                    }
+                }
+
+                groupNames.add(groupName);
+                ordi = false;
+            }
+
+            appendSplitGroups(result, groupNames);
+
+        } else {
+            boolean ordi = ordinal && numberIsWhole; // not ordinal if not whole
+            final List<Long> groups = Utils.splitByModulus(numberLong, 1000000);
+            final List<String> groupNames = new ArrayList<>();
+            for (int i = 0; i < groups.size(); ++i) {
+                final long z = groups.get(i);
+                if (z == 0) {
+                    continue; // skip 000000 groups
+                }
+
+                String groupName;
+                if (z < 1000) {
+                    groupName = subThousand(z, i == 0 && ordi);
+                } else {
+                    groupName = subThousand(z / 1000, false) + " thousand";
```

Comment:
I think these are untranslated: "thousand", "thousandth", "th", "zero", "point" and maybe others